chainer/functions/connection/embed_id.py

```diff
@@ -41,6 +41,10 @@ def forward(self, inputs):
             return xp.where(
                 mask[..., None], 0, W.take(xp.where(mask, 0, x), axis=0)),
 
+        if not issubclass(type(W), type(x)):
```

Comment:
Thanks for the review. You mean something like this right?
```
>>> W = np.array([[1,2], [3,4]])
>>> x = np.matrix("0; 1")
>>> W.take(x) # works
array([[1],
       [2]])
>>> issubclass(type(W), type(x))
False
>>>
```

I will fix it soon to accept this case as well and add unit tests in a couple of days.